plutus-tx/src/PlutusTx/Builtins/Internal.hs

```diff
@@ -1058,3 +1058,7 @@ expModInteger b e m =
     BuiltinSuccess bs -> toInteger bs
     BuiltinSuccessWithLogs logs bs -> traceAll logs $ toInteger bs
 {-# OPAQUE expModInteger #-}
+
+caseInteger :: Integer -> [a] -> a
+caseInteger i b = b !! fromIntegral i
+{-# OPAQUE caseInteger #-}
```

Comment:
This has an issue that `fromIntegral` can under/overflow, unlike the UPLC `case`. Probably doesn't matter, but worth mentioning.

Thanks for calling it `caseInteger` and not `caseInt` by the way.

Reply:
`Int` is a separate type, I don't think that saving 4 characters would justify the potential for confusion.